api/common/v1alpha1/targetref.go

```diff
@@ -88,28 +89,27 @@ type TargetRef struct {
 
 	// Kind of the referenced resource
 	// +kubebuilder:validation:Enum=Mesh;MeshSubset;MeshGateway;MeshService;MeshExternalService;MeshMultiZoneService;MeshServiceSubset;MeshHTTPRoute;Dataplane
-	Kind TargetRefKind `json:"kind,omitempty"`
+	Kind TargetRefKind `json:"kind"`
 	// Name of the referenced resource. Can only be used with kinds: `MeshService`,
 	// `MeshServiceSubset` and `MeshGatewayRoute`
-	Name string `json:"name,omitempty"`
+	Name *string `json:"name,omitempty"`
 	// Tags used to select a subset of proxies by tags. Can only be used with kinds
 	// `MeshSubset` and `MeshServiceSubset`
-	Tags map[string]string `json:"tags,omitempty"`
+	Tags *map[string]string `json:"tags,omitempty"`
 	// Mesh is reserved for future use to identify cross mesh resources.
-	Mesh string `json:"mesh,omitempty"`
+	Mesh *string `json:"mesh,omitempty"`
 	// ProxyTypes specifies the data plane types that are subject to the policy. When not specified,
 	// all data plane types are targeted by the policy.
-	// +kubebuilder:validation:MinItems=1
```

Comment:
@lukidzi does this need to be validated in code for `Mesh` and `MeshSubset`?

Reply:
this is your original PR: https://github.com/kumahq/kuma/pull/8868/files#r1457386523

Reply:
Yes, it makes sense only for this since when you select something more specific you won't have a mix of sidecar/gateway

Reply:
If there is a value defined, it has have at least one value, and just omit when no variable

Reply:
added validation for these two types